src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSpark.java

```diff
@@ -0,0 +1,86 @@
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadsWriteFormat;
+
+import java.io.IOException;
+import java.util.List;
+
+@CommandLineProgramProperties(summary = "Print reads from the input BAM", oneLineSummary = "Print reads from the input BAM", programGroup = SparkProgramGroup.class)
+public final class PrintReadsSpark extends SparkCommandLineProgram {
+
+    public static final String SHARDED_OUTPUT_LONG_ARG = "shardedOutput";
+    public static final String SHARDED_OUTPUT_SHORT_ARG = "shardedOutput";
+
+    private static final long serialVersionUID = 1L;
+
+    @ArgumentCollection
+    public ReadInputArgumentCollection readArguments= new RequiredReadInputArgumentCollection();
+
+    @ArgumentCollection
+    public IntervalArgumentCollection intervalArgumentCollection = new OptionalIntervalArgumentCollection();
+
+    @Argument(doc = "uri for the output file: a local file path",
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            optional = false)
+    public String output;
+
+    @Argument(doc = "If specified, shard the output bam", shortName = SHARDED_OUTPUT_SHORT_ARG, fullName = SHARDED_OUTPUT_LONG_ARG, optional = true)
+    public boolean shardedOutput = false;
+
+    @Override
+    protected void runPipeline(final JavaSparkContext ctx) {
+        if ( readArguments.getReadFilesNames().size() != 1 ) {
+            throw new UserException("This tool only accepts a single bam/sam/cram as input");
+        }
+        final String bam = readArguments.getReadFilesNames().get(0);
+
+        final ReadsSparkSource readSource = new ReadsSparkSource(ctx);
+        final SAMFileHeader readsHeader = ReadsSparkSource.getHeader(ctx, bam);
+
+        if (readsHeader.getSortOrder() != SAMFileHeader.SortOrder.coordinate){
+            //https://github.com/broadinstitute/hellbender/issues/929
+            throw new UserException("PrintReadsSpark: Only coordinate-sorted files are currently supported");
+        }
+        final JavaRDD<GATKRead> reads;
+        /*
+         * If no intervals are given, we want all reads, mapped and unmapped.
+         */
+        if (intervalArgumentCollection.intervalsSpecified()) {
+            final List<SimpleInterval> intervals =  intervalArgumentCollection.getIntervals(readsHeader.getSequenceDictionary());
+            reads = readSource.getParallelReads(bam, intervals);
+        } else {
+            reads = readSource.getParallelReads(bam, null);
+        }
+
+        try {
+            ReadsSparkSink.writeReads(ctx, output, reads, readsHeader, shardedOutput ? ReadsWriteFormat.SHARDED : ReadsWriteFormat.SINGLE);
+        } catch (final IOException e) {
+            throw new GATKException("unable to write bam: " + e);
+        }
+    }
+
+    @Override
+    protected String getProgramName() {
+        return PrintReadsSpark.class.getSimpleName();
```

Comment:
yes, i thought this was in the test class.

thanks!

On Tue, Sep 29, 2015 at 9:52 AM, Tom White <notifications@github.com> wrote:

> In
> src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSpark.java
> <https://github.com/broadinstitute/hellbender/pull/931#discussion_r40673097>
> :
>
> > +            final List<SimpleInterval> intervals =  intervalArgumentCollection.getIntervals(readsHeader.getSequenceDictionary());
> > +            reads = readSource.getParallelReads(bam, intervals);
> > +        } else {
> > +            reads = readSource.getParallelReads(bam, null);
> > +        }
> > +
> > +        try {
> > +            ReadsSparkSink.writeReads(ctx, output, reads, readsHeader, shardedOutput ? ReadsWriteFormat.SHARDED : ReadsWriteFormat.SINGLE);
> > +        } catch (final IOException e) {
> > +            throw new GATKException("unable to write bam: " + e);
> > +        }
> > +    }
> > +
> > +    @Override
> > +    protected String getProgramName() {
> > +        return PrintReadsSpark.class.getSimpleName();
>
> It's the current object (this). So return getClass().getSimpleName();
> should work.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/hellbender/pull/931/files#r40673097>.
>